src/mono/mono/metadata/native-library.c

```diff
@@ -1336,6 +1340,16 @@ lookup_pinvoke_call_impl (MonoMethod *method, MonoLookupPInvokeStatus *status_ou
 	}
 #endif
 
+#if defined(ENABLE_NETCORE) && !defined(NO_GLOBALIZATION_SHIM)
+	if (strcmp(new_scope, "libSystem.Globalization.Native") == 0) {
+		const void* method_impl = GlobalizationResolveDllImport(new_import);
+		if (method_impl)
+		{
+			return (gpointer)method_impl;
+		}
+	}
+#endif
+
```

Comment:
nit: can we pull this out into a `static` function called something like "default_resolve_dllimport" so that the correspondence to the CoreCLR implementation is more evident?

also: is this PR intended to add support for the `PINVOKE_OVERRIDE` property on Mono too? or will that be separate/future work?
see `monovm_initialize` (called by Mono's `coreclr_initialize` hosting API implementation) and its `parse_properties` helper function for the right place to wire us up.

Reply:
> can we pull this out into a static function called something like "default_resolve_dllimport"

NP. Will do.

>  is this PR intended to add support for the PINVOKE_OVERRIDE property on Mono too? or will that be separate/future work?

I would prefer to make that a separate PR. Mostly because it can be separate.
The current PR is already making me a bit uncomfortable because of the number of areas it is affecting.

Reply:
Done in the recent changes